src/main.rs

```diff
@@ -122,16 +122,26 @@ impl MiniserveConfig {
         };
 
         let route_prefix = match (args.route_prefix, args.random_route_prefix) {
-            (Some(prefix), _) if prefix.starts_with('/') => prefix,
-            (Some(prefix), _) => format!("/{}", prefix),
+            (Some(mut prefix), _) => match (prefix.starts_with("/"), prefix.ends_with("/")) {
+                (true, true) => {
+                    prefix.pop();
+                    prefix
+                }
+                (true, false) => prefix,
+                (false, true) => {
+                    prefix.pop();
+                    format!("/{}", prefix)
+                }
+                (false, false) => format!("/{}", prefix),
+            },
```

Comment:
```suggestion
            (Some(prefix), _) => format!("/{}", prefix.trim_matches('/')),
```

I had to go through `&str` docs to see if I can simplify this and I've just learned a new nice method :)